lib/salus/scanners/report_gradle_deps.rb

```diff
@@ -0,0 +1,46 @@
+require 'salus/scanners/base'
+
+# Report Java library usage
+
+module Salus::Scanners
+  class ReportGradleDeps < Base
+    UNKNOWN_VERSION = ''.freeze
+
+    def run
+      shell_return = run_shell("/home/bin/parse_gradle_deps")
+
+      if !shell_return.success?
+        report_error(shell_return.stderr)
+        return
+      end
+
+      dependencies = nil
+      begin
+        dependencies = JSON.parse(shell_return.stdout)
+      rescue JSON::ParserError
+        err_msg = "Could not parse JSON returned by /home/bin/parse_gradle_deps's stdout!"
+        report_stderr(err_msg)
+        report_error(err_msg)
```

Comment:
You should `return` here.  
Otherwise, `dependencies` will stay `nil`, and then when `dependencies.each` runs, there will be a `each not defined on nil exception`.